src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -377,28 +423,36 @@ impl StaticHeader {
                     }
                 }
                 (Err(loc_1), Ok(loc_2)) => {
-                    if loc_2.is_some() {
-                        BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
-                        Ok(loc_2)
+                    // Re-write copy 1
+                    if let Some(loc_2) = loc_2 {
+                        bda_write_check(f, &buf_loc_2, MetadataLocation::First, loc_2)
                     } else {
                         // Location 2 doesn't have a signature, but location 1 did, but it got an error,
                         // lets return the error instead as this appears to be a stratis device that
                         // has gotten in a bad state.
                         Err(loc_1)
                     }
                 }
-                (Err(_), Err(_)) => {
-                    let err_str = "Appeared to be a Stratis device, but no valid sigblock found";
-                    Err(StratisError::Engine(ErrorEnum::Invalid, err_str.into()))
+                (None, None) => Ok(None),
+                (Some(loc_1), None) => {
+                    // Copy 1 has valid Stratis BDA, copy 2 has no magic, re-write copy 2
+                    BDA::write(f, &buf_loc_1, MetadataLocation::Second)?;
+                    Ok(Some(loc_1))
+                }
+                (None, Some(loc_2)) => {
```

Comment:
Here's the compilation failure from the stratisd Jenkins test:
```
error[E0308]: mismatched types
   --> src/engine/strat_engine/backstore/metadata/bda.rs:436:18
    |
436 |                 (None, None) => Ok(None),
    |                  ^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected type `std::result::Result<std::option::Option<engine::strat_engine::backstore::metadata::bda::StaticHeader>, stratis::errors::StratisError>`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> src/engine/strat_engine/backstore/metadata/bda.rs:436:24
    |
436 |                 (None, None) => Ok(None),
    |                        ^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected type `std::result::Result<std::option::Option<engine::strat_engine::backstore::metadata::bda::StaticHeader>, stratis::errors::StratisError>`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> src/engine/strat_engine/backstore/metadata/bda.rs:437:18
    |
437 |                 (Some(loc_1), None) => {
    |                  ^^^^^^^^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected type `std::result::Result<std::option::Option<engine::strat_engine::backstore::metadata::bda::StaticHeader>, stratis::errors::StratisError>`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> src/engine/strat_engine/backstore/metadata/bda.rs:437:31
    |
437 |                 (Some(loc_1), None) => {
    |                               ^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected type `std::result::Result<std::option::Option<engine::strat_engine::backstore::metadata::bda::StaticHeader>, stratis::errors::StratisError>`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> src/engine/strat_engine/backstore/metadata/bda.rs:442:18
    |
442 |                 (None, Some(loc_2)) => {
    |                  ^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected type `std::result::Result<std::option::Option<engine::strat_engine::backstore::metadata::bda::StaticHeader>, stratis::errors::StratisError>`
               found type `std::option::Option<_>`

error[E0308]: mismatched types
   --> src/engine/strat_engine/backstore/metadata/bda.rs:442:24
    |
442 |                 (None, Some(loc_2)) => {
    |                        ^^^^^^^^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected type `std::result::Result<std::option::Option<engine::strat_engine::backstore::metadata::bda::StaticHeader>, stratis::errors::StratisError>`
               found type `std::option::Option<_>`

error: aborting due to 6 previous errors
```

Reply:
Thanks! I failed the rebase pretty badly apparently. :laughing: